contracts/MaticTokenVesting.sol

```diff
@@ -80,14 +84,20 @@ contract TokenVesting is Ownable {
     }
 
     function removeVesting(uint256 _vestingId) public onlyOwner {
-        require(vestings[_vestingId].beneficiary != address(0x0) && !vestings[_vestingId].released);
-        vestings[_vestingId].released = true;
+        Vesting storage vesting = vestings[_vestingId];
+        require(vesting.beneficiary != address(0x0), "INVALID_VESTING_ID");
+        require(!vesting.released , "VESTING_ALREADY_RELEASED");
+        vesting.released = true;
+        tokensToVest -= vesting.amount;
+        emit TokensVestingRemoved(_vestingId);
     }
 
     function addVesting(address _beneficiary, uint256 _releaseTime, uint256 _amount) public onlyOwner {
-        require(_beneficiary != address(0x0), "ADD_BENEFICIARY");
+        require(_beneficiary != address(0x0), "INVALID_BENEFICIARY_ADDRESS");
+        require(maticToken.balanceOf(address(this)) >= tokensToVest + _amount, INSUFFICIENT_BALANCE);
```

Comment:
I think this will create `deadlock`. We are adding most vesting periods in constructor and when we deploy this contract, we may not have no tokens in contract.

Reply:
Got it. Didn't know test data in the constructor will be replaced by actual vestings. I thought it's just there to test something and will be removed.